Main.hs

```diff
@@ -110,12 +110,25 @@ isOp (Hs.UnQual _ Hs.Symbol{}) = True
 isOp (Hs.Special _ Hs.Cons{})  = True
 isOp _                         = False
 
+freshString :: String -> TCM String
+freshString s = do
+  ctx <- getContextNames
+  -- prettyShow <$> freshName_ s
+  -- ^ this returns 'x' even when context contains 'x'??
```

Comment:
Am I using `freshName_` improperly here?

Reply:
Try `prettyTCM` instead of `prettyShow`.

Reply:
Aha, it works!